pretyping/coercion.ml

```diff
@@ -521,6 +521,13 @@ let rec inh_conv_coerce_to_fail ?loc env evd rigidonly v t c1 =
 
 (* Look for cj' obtained from cj by inserting coercions, s.t. cj'.typ = t *)
 let inh_conv_coerce_to_gen ?loc resolve_tc rigidonly env evd cj t =
+  let evd, cj =
+    let evd, uj_type = Evarsolve.refresh_universes
+        ~status:Evd.univ_flexible ~onlyalg:true (Some false)
+        env evd cj.uj_type
+    in
+    evd, {cj with uj_type }
+  in
```

Comment:
I don't know if this is the right place to put this, we could also go for `Pretyping.inh_conv_coerce_to_tycon` (with possibility of missing other callers).
Here seems OK though.

Reply:
It sounds quite strong to always refresh, while leq can handle cj.uj_type being an algebraic.